lib/webapp.js

```diff
@@ -63,7 +63,47 @@ function DataserviceContext(serviceDefinition, serviceConfiguration,
   this.plugin = pluginContext;
   this.logger = global.COM_RS_COMMON_LOGGER.makeComponentLogger(
     pluginContext.pluginDef.identifier + ":" + serviceDefinition.name);
+  setLoggerMessages(this.logger, 
+    pluginContext.server.config.user.logLanguage,
+    pluginContext.pluginDef.location);
 }
+
+function setLoggerMessages(logger, logLanguage, logLocation) {
```

Comment:
How about name this "createDataserviceLogger" and have it return a logger to assign to this.logger.
Since this is all synchronous, you do not need to hack in `logger._messages`, just create the logger AFTER getting the messages, and return the logger above.